README.md

```diff
@@ -314,52 +315,95 @@ module.exports = {
 };
 ```
 
-## Configuration
+## Underreact configuration file
 
-**No configuration is necessary to get started.** On most production projects you'll want to set at least a few of the "Basic configuration" options. "Advanced configuration" options shouldn't be necessary in *most* cases ... but you never know.
+Underreact expects a `underreact.config.js` file to exist at the root of your project. Please note that **No configuration is necessary to get started.** On most production projects you'll want to set at least a few of the [configuration object](#configuration-object-properties) properties.
 
-### siteBasePath
+Your `underreact.config.js` can look like either of the below:
 
-Type: `string`. Default: `''`.
+- **Exporting a Function (Preferred)**: You can export a function which would then be used as a factory method for your [configuration object](#configuration-object-properties). This function is called the following parameter properties of an object:
 
-Root-relative path to the base directory on the domain where the site will be deployed.
+```javascript
+// underreact.config.js
+/**
+ * @param {Object} opts 
+ * @param {Webpack} opts.webpack - The Webpack object.
+ * @param {'start'|'build'|'serve-static'} opts.command - The current command Underreact is following.
+ * @param {'production'|'development'} opts.mode - The current mode of Underreact.
+ */
+module.exports = function({ webpack, command, mode }) {
+  return {/*Underreact configuration object*/}
+}
+```
 
-There's a good chance your app isn't at the root of your domain. So this option represents the path of your site *within* that domain.
+- **Exporting an Object**: You can also directly export the [configuration object](#configuration-object-properties). This is a great way to start experimenting with Underreact's configuration. For example in the code below we simply want to modify the `siteBasePath`:
 
-For example, if your app is at `https://www.special.com/ketchup/*`, you should set `siteBasePath: ketchup`.
+```javascript
+// underreact.config.js
+module.exports = {
+   siteBasePath: 'fancy'
+}
+```
 
-### stylesheets
+## Configuration Object Properties
```

Comment:
The convention I've been going with is to capitalize only the first word of headings (or, of course, other words that need to be capitalized, like names).